File: app/src/main/kotlin/data/Person.kt
data class Person(
    val id: String = UUID.randomUUID().toString(),
    val firstName: String,
    val lastName: String,
    val birthDate: LocalDate,
    val deathDate: LocalDate? = null,
    val photoUrl: String? = null
)

sealed class Screen {
    object PeopleList : Screen()
    data class PersonDetails(val personId: String) : Screen()
    object AddPerson : Screen()
    data class EditPerson(val personId: String) : Screen()
}

File: app/models/person.ts
export class Person {
    id: string;
    firstName: string;
    lastName: string;
    birthDate: Date;
    deathDate?: Date;
    photoUrl?: string;

    constructor(firstName: string, lastName: string, birthDate: Date) {
        this.id = crypto.randomUUID();
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
    }
}